### PR TITLE
Auto add an Index on spid for ServiceProviderAware DAOs.

### PR DESCRIPTION
**Pull request by jlhughes**
Also use MDAO.GET_MDAO_CMD to attempt to get proxied MDAO to add Indexes on.

### CODE REVIEW

File: src/foam/dao/EasyDAO.js [hunk 1]
```diff
@@ -221,9 +225,26 @@ foam.CLASS({
         }
 
         if ( getServiceProviderAware() ) {
-          delegate = new foam.nanos.auth.ServiceProviderAwareDAO.Builder(getX())
+         delegate = new foam.nanos.auth.ServiceProviderAwareDAO.Builder(getX())
```
Comment (jlhughes) [inline src/foam/dao/EasyDAO.js]:
```suggestion
          delegate = new foam.nanos.auth.ServiceProviderAwareDAO.Builder(getX())
```

File: src/foam/dao/EasyDAO.js [hunk 2]
```diff
@@ -1026,14 +1047,14 @@ model from which to test ServiceProvider ID (spid)`,
         return this;
       },
       javaCode: `
-        if ( getInnerDAO() == null && getMdao() != null ) {
-          ((foam.dao.MDAO) getMdao()).addIndex(props);
-        } else {
-          Logger logger = (Logger) getX().get("logger");
-          if ( logger == null ) {
-            logger = new foam.nanos.logger.StdoutLogger();
+        if ( getMdao() != null ) {
```
Comment (lchanmann) [inline src/foam/dao/EasyDAO.js]: Are we no longer want to log when the nspec tries to add index to wrapper easy DAO? eg. no log for when adding index in localUserDAO?

Comment (jlhughes) [reply src/foam/dao/EasyDAO.js]: The  cmd(MDAO.GET_MDAO_CMD) will traverse the delegates and return if it his another EasyDAO or a real MDAO, so we can attempt to add indexes on nspecs such as localUserDAO, and if we can't then it warns.

Comment (lchanmann) [reply src/foam/dao/EasyDAO.js]: but we don't setMdao() when there is an innerDAO, do we? so getMdao() here will be null, no?

Comment (jlhughes) [reply src/foam/dao/EasyDAO.js]: yeah, I can just call the cmd on the delegate, no concern for the mdao, good call.

**Review by lchanmann — APPROVED** [review]
LGTM

**Pull request merged**